test/qos2.js

```diff
@@ -209,6 +209,39 @@ test('subscribe QoS 0, but publish QoS 2', function (t) {
   })
 })
 
+test('subscribe QoS 1, but publish QoS 2', function (t) {
+  var broker = aedes()
+  var publisher = connect(setup(broker))
+  var subscriber = connect(setup(broker))
+  var expected = {
+    cmd: 'publish',
+    topic: 'hello',
+    payload: Buffer.from('world'),
+    qos: 1,
+    dup: false,
+    length: 14,
+    retain: false
+  }
+
+  subscribe(t, subscriber, 'hello', 1, function () {
+    subscriber.outStream.once('data', function (packet) {
+      delete packet.messageId
+      t.deepEqual(packet, expected, 'packet must match')
+      t.end()
+    })
```

Comment:
Can you add some checks on the follow up packets to verify that the full QoS flow is respected?

Reply:
Sure.

Reply:
@mcollina  I was just checking. You have helper functions in place to check the QoS flow.
 https://github.com/mcollina/aedes/blob/ca42eb0164e08538cabf2ddaa087ef1ff4b9dc8e/test/qos2.js#L11-L46 and https://github.com/mcollina/aedes/blob/ca42eb0164e08538cabf2ddaa087ef1ff4b9dc8e/test/helper.js#L77-L96